File: 第十次/10-4/Student.cpp
#include"Student.h"

Student::Student() {
	name = "Avrill";
	birthDay.setMonth(1);
	birthDay.setDay(1);
	birthDay.setYear(2000);
	score = 100;
}
Student::Student(string newName, Date newBirthDay, int newScore) {
	setName(newName);
	setScore(newScore);
	birthDay = newBirthDay;
}

void Student::setName(string newName) {
	name = newName;
}
string Student::getName() const {
	return name;
}

void Student::setDate(int mon, int day, int year) {
	birthDay.setMonth(mon);
	birthDay.setDay(day);
	birthDay.setYear(year);
}
Date Student::getDate() const {
	return birthDay;
}

void Student::setScore(int newScore) {
	score = newScore;
}
int Student::getScore() const {
	return score;
}

void Student::print() {
	cout << name << " ";
	birthDay.print();
	cout << " " << score << endl;
}

File: Python/StudentCard_Composition/Student.py
from Date import Date

class Student:
    def __init__(self, name, month, day , year , score):
        self.__name = name
        self.__date = Date(month, day, year)
        self.__score = score

    def setName(self, name):
       self.__name = name
    def setDate(self, mon, day, year):
        self.__date.setMonth(mon)
        self.__date.setDay(day)
        self.__date.setYear(year)
    def setScore():
        self.__score = score

    def getName(self):
        return self.__name
    def getDate(self):
      return self.__date
    def getScore(self):
        return self.__score
    def toString(self):
        print(self.__name, self.__date.toString(), self.__score )